lib/ruby_lsp/ruby_lsp_rails/extension.rb

```diff
@@ -14,6 +14,9 @@ class Extension < ::RubyLsp::Extension
 
       sig { override.void }
       def activate
+        ::RubyLsp::Requests::Hover.add_listener(RubyLsp::Rails::Hover)
```

Comment:
Since we only register classes and not instances, I wonder if this could be moved in the Listener itself:

```ruby
class RubyLsp::Rails::CodeLens < ::RubyLsp::Listener
  listen_to ::RubyLsp::Requests::CodeLens

  # ...
end
```

Reply:
That's an interesting approach, but it would lead to the same problem we're trying to avoid in this PR.

In order to disable an extension and avoid registering the listeners defined by it, we need the registration to happen in `activate`.

Basically, if the server extension is disabled, we just don't invoke `activate`. If the listener classes register themselves, then that will happen automatically when we require the `Extension` file and we'll have no way of stopping it.